samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemFactory.java

```diff
@@ -31,20 +35,28 @@
  * Initial draft of in-memory {@link SystemFactory}. It is test only and not meant for production use right now.
  */
 public class InMemorySystemFactory implements SystemFactory {
-  private static final InMemoryManager MEMORY_MANAGER = new InMemoryManager();
+  private static Map<Integer, InMemoryManager> jobToManager = new HashMap<>();
 
   @Override
   public SystemConsumer getConsumer(String systemName, Config config, MetricsRegistry registry) {
-    return new InMemorySystemConsumer(MEMORY_MANAGER);
+    jobToManager.putIfAbsent(parseJobId(config), new InMemoryManager());
```

Comment:
Can we roll this into the `parseJobId` method and maybe rename that to reflect its purpose of creating a manager from config along with the attribute that its keyed on?
E.g. `getManagerFor...` or `getManagerFrom...`

Reply:
since I have configured a new parameter I now call this method getOrDefaultInMemoryManagerByTestId